File: utils/counter.py
class Counter:
  def __init__(self, n):
    self.n = n
    self.current_cnt = n
  
  def update(self):
    self.current_cnt -= 1
    if self.current_cnt <= 0:
      self.current_cnt = self.n
      return True
    return False

File: app/app/src/main/java/com/example/demoiot/Counter.java
package com.example.demoiot;

public class Counter {
    private final int n;
    private int currentCnt;

    public Counter(int n) {
        this.n = n;
        this.currentCnt = n;
    }

    public boolean update() {
        this.currentCnt -= 1;
        if (this.currentCnt <= 0) {
            this.currentCnt = this.n;
            return true;
        }
        return false;
    }

    public void reset(){
        this.currentCnt = this.n;
    }
}
